pyproject.toml

```diff
@@ -71,16 +71,23 @@ strict = true
 lines_between_types = 1
 lines_after_imports = 2
 multi_line_output = 5
-known_first_party = "mesonpy"
+known_first_party = 'mesonpy'
 
 
 [tool.coverage.html]
 show_contexts = true
 
 
 [tool.pytest.ini_options]
-minversion = "6.0"
-addopts = ["-ra", "--strict-markers", "--strict-config"]
-log_cli_level = "info"
-norecursedirs = "tests/packages/*"
-testpaths = ["tests"]
+minversion = '6.0'
+addopts = ['-ra', '--strict-markers', '--strict-config']
+log_cli_level = 'info'
+norecursedirs = 'tests/packages/*'
+testpaths = ['tests']
+xfail_strict = true
+filterwarnings = [
+  'error',
+]
+markers = [
+  'isolated: Isolated builds',
+]
```

Comment:
I don't see the point of adding more undiscoverable pytest magic. The tests run just fine on conda and shaving off a couple of seconds from the test run does not seem worth the trouble.

Reply:
To the best of my knowledge, this is a new test and has never been run on conda? When I've added similar tests to conda in the past, they generally don't work, and that's okay, since you don't do isolated builds on conda in the first place. If it does work, there's no harm done, it just adds a bit of organization.

This is not "undiscoverable", you can run `pytest --markers` and see all the markers, with descriptions - including the ones like `parametrize` and `skipif` which are already being used. Normal users shouldn't have to use this, as well, it just makes it easy for conda and maybe other packagers to skip this if they don't work.

In fact, we could trade `pep518` as a fixture for `pytest.mark.isolated`, if you want - marking it with isolated could auto-add the fixture (the fixture isn't used inside the test body, so it's fairly natural to do so).

I just changed the marking style a little for this iteration, and can change more later if you still don't like it.

Reply:
Tests are run on conda. This PR turns on warnings that break another test I recently added, however, the `test_pep518.py` tests run just fine there.

Reply:
Where are they run on conda? I'm fine to drop this then.

Reply:
See the "sage / linux / linux (conda-forge, minimal-develop)" job.